flocker/node/agents/test/test_blockdevice.py

```diff
@@ -3136,27 +3172,30 @@ def _verify_volume_size(self, requested_size, expected_volume_size):
         self.api.attach_volume(
             volume.blockdevice_id, attach_to=self.this_node,
         )
-        # Reload the volume using ``IBlockDeviceAPI.list_volumes`` in
-        # case the implementation hasn't verified that the requested
-        # size has actually been stored.
-        volume = get_blockdevice_volume(self.api, volume.blockdevice_id)
 
-        device_path = self.api.get_device_path(volume.blockdevice_id).path
+        def validate(volume):
+            # Reload the volume using ``IBlockDeviceAPI.list_volumes`` in
+            # case the implementation hasn't verified that the requested
+            # size has actually been stored.
+            volume = get_blockdevice_volume(self.api, volume.blockdevice_id)
 
-        command = [b"/bin/lsblk", b"--noheadings", b"--bytes",
-                   b"--output", b"SIZE", device_path.encode("ascii")]
-        command_output = check_output(command).split(b'\n')[0]
-        device_size = int(command_output.strip().decode("ascii"))
-        if self.device_allocation_unit is None:
-            expected_device_size = expected_volume_size
-        else:
-            expected_device_size = allocated_size(
-                self.device_allocation_unit, expected_volume_size
+            device_path = self.api.get_device_path(volume.blockdevice_id).path
+
+            command = [b"/bin/lsblk", b"--noheadings", b"--bytes",
+                       b"--output", b"SIZE", device_path.encode("ascii")]
+            command_output = check_output(command).split(b'\n')[0]
+            device_size = int(command_output.strip().decode("ascii"))
+            if self.device_allocation_unit is None:
+                expected_device_size = expected_volume_size
+            else:
+                expected_device_size = allocated_size(
+                    self.device_allocation_unit, expected_volume_size
+                )
```

Comment:
Is `expected_device_size` expected to change on loops? If not, please extract from the function.